x/bank/types/supply.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	"fmt"
+
+	proto "github.com/gogo/protobuf/proto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// SupplyI defines an inflationary supply interface for modules that handle
+// token supply.
+// Deprecated.
+type SupplyI interface {
+	proto.Message
+
+	GetTotal() sdk.Coins
+	SetTotal(total sdk.Coins)
+
+	Inflate(amount sdk.Coins)
+	Deflate(amount sdk.Coins)
+
+	String() string
+	ValidateBasic() error
+}
+
+// Implements Supply interface
+// Deprecated.
+var _ SupplyI = (*Supply)(nil)
+
+// NewSupply creates a new Supply instance
+func NewSupply(total sdk.Coins) *Supply {
+	return &Supply{total}
+}
+
+// DefaultSupply creates an empty Supply
+func DefaultSupply() *Supply {
+	return NewSupply(sdk.NewCoins())
+}
+
+// SetTotal sets the total supply.
+func (supply *Supply) SetTotal(total sdk.Coins) {
```

Comment:
the other alternative would be to move this file to `x/bank/legacy/v040`, but then we would need to duplicate the proto-generated `Supply` struct. I went for no duplication here.